main.cpp

```diff
@@ -23,7 +23,7 @@ bool is_permutation(string str1, string str2) {
 }
 
 struct Node {
-    Node* next;
+    Node* next = NULL;
```

Comment:
I know I explicitly said 'NULL', but I meant nullptr :)
Not a big deal in this case, but I think it's best to get use to use it.